model/publishing/delivery/tasks/DeployTestEnvironments.php

```diff
@@ -160,18 +156,38 @@ private function getQtiTestPackageFile(core_kernel_classes_Resource $delivery):
     {
         $packagePath = $delivery->getOnePropertyValue($this->getProperty(TestBackupService::PROPERTY_QTI_TEST_BACKUP_PATH));
         $packagePath = (string)  $packagePath;
-        
+
         return $this->getServiceLocator()
             ->get(FileSystemService::SERVICE_ID)
             ->getDirectory(TestBackupService::FILESYSTEM_ID)
             ->getFile($packagePath);
     }
 
+    private function getDeliveryPropertyList(\core_kernel_classes_Resource $delivery): array
+    {
+        $propertyList = [];
+        foreach ($this->getPublishingDeliveryService()->getSyncFields() as $deliveryProperty) {
+            $value = $delivery->getOnePropertyValue($this->getProperty($deliveryProperty));
+            if ($value instanceof \core_kernel_classes_Resource) {
+                $value = $value->getUri();
+            }
+            $propertyList[$deliveryProperty] = (string) $value;
+        }
+        $propertyList[PublishingDeliveryService::ORIGIN_DELIVERY_ID_FIELD] = $delivery->getUri();
```

Comment:
can you move this to the $propertyList declaration?

Reply:
It's a bit of defensive programming, making sure it's not overwritten by properties read from DB.

Reply:
Ah, I see, then could you add a commentary there, please?